src/Language/DifferentialDatalog/Compile.hs

```diff
@@ -624,7 +624,11 @@ simplifyDelayedRels d =
            delayed_rels d'
     where
     mk_delayed_rel_name :: (String, Delay) -> String
-    mk_delayed_rel_name (rel, del) = "__" ++ show (delayDelay del) ++ "_" ++ rel
+    mk_delayed_rel_name (rel, del) = scoped rel_scope $ "__" ++ show (delayDelay del) ++ "_" ++ rel_name
```

Comment:
Can you please add a documentation line in the reference manual that some identifiers are reserved?